src/rebar_packages.erl

```diff
@@ -243,6 +243,7 @@ parse_checksum(Checksum) ->
 
 update_package(Name, RepoConfig=#{name := Repo}, State) ->
     ?MODULE:verify_table(State),
+    ?DEBUG("Getting package ~ts resource from repo ~ts", [Name, maps:get(repo_url, RepoConfig, undefined)]),
```

Comment:
I don't know that it makes the most sense to add this debug statement here. This call is internal and can get a lot of calls, with some of this information not actually being accurate. The update_package call here for example gets a debug message when it fails in `rebar_prv_upgrade` which refers to the repo's `name` and not the URL. That's because internally, the call gets the package definition and not the actual package, and the URL is not just the base value configured here, but has a subpath that may be organisation specific as well.

This is likely to be more misleading than it helps if something goes wrong. It might make more sense to have something like the following, maybe:

```suggestion
    ?DEBUG("Getting definition for package ~ts resource from repo ~ts", [Name, maps:get(name, RepoConfig)]),
```

Reply:
The same as below, `RepoConfig#{repo_url := Value}` here is use for build the URL for making request in `r3_hex_repo` as a prefix. The question is its name, `repo_url` is just a prefix, not a hole url for a request. But it's still useful in the debug output for people who want to check the http requests detail. Maybe describe it `repo_url` directly won't make misleadings. So how about 
```suggestion
    ?DEBUG("Getting package ~ts resource from repo_url ~ts", [Name, maps:get(repo_url, RepoConfig)]),
```

Reply:
I think that can work. One of the things I might do while merging is to follow the precedent I started setting in https://github.com/erlang/rebar3/pull/2375 and https://github.com/erlang/rebar3/pull/2393 to make it more obvious which settings are active at download time. I'll print the configured map with private fields elided and use some caching to make sure they're not printed more than once per run or something. I don't yet know which shape I want it to take, so I'll avoid blocking your PR on this.

Reply:
Thanks for the review. Debugging format won't affect most uses I guess. But they are still worth to be considered and well-designed.